packages/core/src/modules/connections/ConnectionsModule.ts

```diff
@@ -197,6 +198,91 @@ export class ConnectionsModule {
     return this.connectionService.getAll()
   }
 
+  /**
+   * Allows for the addition of connectionType to the record.
+   *  Either updates or creates an array of string conection types
+   * @param connectionId
+   * @param type
+   * @throws AriesFrameworkError if the record corrisponding to the provided connectionId cannot be found
+   */
+  public async addConnectionType(connectionId: string, type: ConnectionType | string) {
+    const record = await this.findById(connectionId)
+    if (record) {
+      const tags = (record.getTag('connectionType') as string[]) || ([] as string[])
+      record.setTag('connectionType', [type, ...tags])
+      await this.connectionService.update(record)
+    } else {
+      throw new AriesFrameworkError('The provided connectionId did not match a known record')
+    }
+  }
+  /**
+   * Removes the given tag from the given record found by connectionId, if the tag exists otherwise does nothing
+   * @param connectionId
+   * @param type
+   * @throws AriesFrameworkError if the record corrisponding to the provided connectionId cannot be found
+   */
+  public async removeConnectionType(connectionId: string, type: ConnectionType | string) {
+    const record = await this.findById(connectionId)
+    if (record) {
+      const tags = (record.getTag('connectionType') as string[]) || ([] as string[])
+
+      const newTags = tags.filter((value: string) => {
+        if (value != type) return value
+      })
+      record.setTag('connectionType', [...newTags])
+
+      await this.connectionService.update(record)
+    } else {
+      throw new AriesFrameworkError('The provided connectionId did not match a known record')
+    }
+  }
+  /**
+   * Gets the known connection types for the record matching the given connectionId
+   * @param connectionId
+   * @returns An array of known connection types or null if none exist
+   * @throws AriesFrameworkError if the record corrisponding to the provided connectionId cannot be found
+   */
+  public async getConnectionTypes(connectionId: string) {
+    const record = await this.findById(connectionId)
+    if (record) {
+      const tags = record.getTag('connectionType') as string[]
+      return tags || null
+    } else {
+      throw new AriesFrameworkError('The provide connectionId did not match a known record')
+    }
+  }
+  /**
+   * tries to find the target tag on the record corrisponding to provided connectionId
+   * @param connectionId
+   * @param targetTag name of the target tag as a string (case sensitive)
+   * @returns The target tag or null if it does not exist
+   * @throws AriesFrameworkError if the record corrisponding to the provided connectionId cannot be found
+   */
+  public async getTagFromRecord(connectionId: string, targetTag: string) {
```

Comment:
I usually get the tags by doing something like `await agent.connections.getById(connectionId)).getTags()` or `(await agent.connections.getById(connectionId).getTag(targetTag)`.

Would be anything different by calling these new methods?

Reply:
It is the exact same, just got over zealous without realizing that is there already.